src/vs/workbench/contrib/codeEditor/browser/outline/documentSymbolsOutline.ts

```diff
@@ -358,7 +358,7 @@ class DocumentSymbolsOutline implements IOutline<DocumentSymbolItem> {
 			}));
 
 			// feature: update active when cursor changes
-			this._outlineDisposables.add(this._editor.onDidChangeCursorPosition(_ => {
+			this._outlineDisposables.add(this._editor.onDidChangeCursorPositionDeferred(_ => {
```

Comment:
This listener is already async (`TimeoutTimer#cancelAndSet`) and I don't think double debounce is needed

Reply:
This was what I was trying to avoid, deferring the extra clear and timeout calls into the existing timeout, but it didn't give the gains expected.